### PR TITLE
Record Selector styling

### LINKED ISSUE #2104: Implement record selector styling updates
Design: https://www.figma.com/file/xHb5oIqye3fnXtb2heRH34/Styling?node-id=2659%3A20227

### PR DESCRIPTION
**Pull request by seancolsen**
Fixes #2104

## Before this PR

image

## Figma design

image

## After this PR

image

## Notable differences vs Figma

- I used a thicker outline to indicate focus of the current input.
- I adjusted the yellow color of the match highlight slightly to give it more contrast.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by seancolsen** [comment]
(Re-assigning review to Rajat because I see that Pavish currently has several other PRs to review.)

**Comment by pavish** [comment]
@seancolsen The [consistency fixes PR](https://github.com/centerofci/mathesar/pull/2146) contains some styling changes which affect the record selector. Do you think you could base your PR on top of it, and resolve conflicts? This would be much faster than the other way around. Also, I expect the consistency fix PR to be merged tomorrow.

**Comment by seancolsen** [comment]
> base your PR on top of it, and resolve conflicts

Thanks for pointing that out, @pavish. Done.

File: mathesar_ui/src/systems/record-selector/RecordSelectorWindow.svelte [hunk 1]
```diff
@@ -105,15 +93,12 @@
...
-    --z-index__record_selector__focused-input: 6;
-    --z-index__record_selector__shadow-inset: 7;
-    --z-index__record_selector__overlay: 8;
-    --z-index__record_selector__above-overlay: 9;
+    --z-index__record_selector__row-header: 1;
+    --z-index__record_selector__thead: 2;
+    --z-index__record_selector__thead-row-header: 3;
+    --z-index__record_selector__shadow-inset: 4;
+    --z-index__record_selector__overlay: 5;
+    --z-index__record_selector__above-overlay: 6;
```
Comment (rajatvijay) [inline mathesar_ui/src/systems/record-selector/RecordSelectorWindow.svelte:101]: should we add a note for this convention in the styling guide? I still don't think I understand this notation fully.

**Review by rajatvijay — DISMISSED** [review]
@seancolsen I saw 2 weird things happening here: 

1. Layout shift when searching inside the record selector: https://www.loom.com/share/29ca246255cd4ba298fdf9308bca0529
2. In the below screenshot the `Acquisition Date` has no matches but the results are still shown because of the `id` and `Acquisition price` match. It might make sense but it looked a bit weird to me at first look. 
Screenshot 2023-01-05 at 1 02 46 PM

File: mathesar_ui/src/systems/record-selector/RecordSelectorWindow.svelte (hunk 1 above)
Comment (seancolsen) [reply mathesar_ui/src/systems/record-selector/RecordSelectorWindow.svelte:101]: I added https://github.com/centerofci/mathesar-wiki/commit/07d21b7805beca4e4884ef23828ed33abe373628 to explain this convention more. Does this make sense?

**Comment by rajatvijay** [comment]
@seancolsen LGTM 🚀

**Comment by kgodey** [comment]
@rajatvijay I had to dismiss your review to get this merged, please remember to use the "approve" functionality. Otherwise, auto-merge won't work.

**Pull request merged**